File: leetcode-algorithms/038. Count and Say/CountAndSay.cs
public class Solution {
    public string CountAndSay(int n) {
        if(n==1){
            return "1";
        }
        if(n==2){
            return "11";
        }
        string temp = CountAndSay(n-1);
        string result = "";
        int count = 1;
        for(int i=0;i<temp.Length-1;i++){

            if(temp[i]==temp[i+1]){
                count++;
                if(i==temp.Length-2){
                    result+=count.ToString()+temp[i];
                }
            }
             else
            {
                
                result+=count.ToString()+temp[i];
                count = 1;
                if(i==temp.Length-2){
                    result+="1"+temp[i+1];
                }
            }
            
        }
        return result;
    }
}

File: leetcode-algorithms/038. Count and Say/CountAndSay.py
class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if n == 1:
            return "1"
        result = ""
        lasted = list(self.countAndSay(n-1))
        while len(lasted) != 0:
            count = 1
            head = lasted.pop(0)
            while len(lasted) != 0 and head == lasted[0]:
                count += 1
                lasted.pop(0)
            result += "".join([str(count),head])
        return result